service/history/queues/queue_scheduled.go

```diff
@@ -303,10 +304,14 @@ func (p *scheduledQueue) lookAheadTask() {
 // IsTimeExpired checks if the testing time is equal or before
 // the reference time. The precision of the comparison is millisecond.
 func IsTimeExpired(
```

Comment:
I would add a note explaining why we take the task here (to prevent clock skew issues).